### ISSUE TITLE
Eager stack backward fails for mixed tracked and untracked inputs

### ISSUE THREAD

**Issue by shinaoka** [tensor4all/tenferro-rs#1692]
## Summary

`EagerTensor::stack` / `concatenate` fails during reverse-mode compilation when its inputs mix tracked and untracked tensors.

This blocks tensor4all-rs issue tensor4all/tensor4all-rs#623 from pinning tenferro commit `b7b4cbcf4ab31b0aa6c1d3b4fcdd0b6d69702bef`.

## Reproducer

```rust
use tenferro_ad::{EagerRuntime, EagerTensor, Tensor};
use tenferro_cpu::CpuBackend;

let ctx = EagerRuntime::with_cpu_backend(CpuBackend::new())?;
let tracked = EagerTensor::requires_grad_in(
    Tensor::from_vec_col_major(vec![2], vec![1.0_f64, 2.0])?,
    ctx.clone(),
)?;
let constant = EagerTensor::from_tensor_in(
    Tensor::from_vec_col_major(vec![2], vec![3.0_f64, 4.0])?,
    ctx,
)?;

let stacked = EagerTensor::stack(&[&tracked, &constant], -1)?;
let loss = stacked.reduce_sum(Some(&[0, 1]))?;
loss.backward()?;

assert_eq!(tracked.grad()?.unwrap().as_slice::<f64>()?, &[1.0, 1.0]);
```

Observed at `b7b4cbc`:

```text
shape_infer (Compile): invalid argument shapes: input 1 dimension 1 does not match the first input
```

Both inputs tracked succeeds. The failure also reproduces with the tracked and untracked inputs reversed and with insertion axis `0`.

## Likely cause

`EagerTensor::stack` reshapes each operand and calls n-ary `Concatenate`. Active-edge AD prunes the inactive constant edge, but concatenate VJP shape/offset inference still needs complete primal shape metadata for every original operand.

Mixed tracked/untracked operands are otherwise supported as constant feeds to tracked eager operations. The desired fix is to retain/resolve all concatenate operand shapes while pruning only inactive cotangent outputs.

## Regression coverage requested

- mixed tracked/untracked `EagerTensor::stack` in both orders
- insertion axes `0` and `-1`
- mixed-activity direct concatenate with unequal concatenate-axis extents
- exact weighted gradient checks for f64 and Complex64
